engine/src/main/java/org/terasology/network/internal/ClientConnectionHandler.java

```diff
@@ -98,6 +98,15 @@ public void messageReceived(ChannelHandlerContext ctx, MessageEvent e) {
 
         // Handle message
         NetData.NetMessage message = (NetData.NetMessage) e.getMessage();
+        String errorMessage = message.getServerInfo().getErrorMessage();
+        if (errorMessage != null && !errorMessage.isEmpty()) {
+            synchronized (joinStatus) {
+                joinStatus.setErrorMessage(errorMessage);
```

Comment:
At some point context is going to have to be made available to this class to fix this line in the constructor, probably by passing a context into the constructor.
`
        this.moduleManager = CoreRegistry.get(ModuleManager.class);
`
For now, I think it's sufficient to add a todo to the effect of:
`
// TODO: implement translation of error messages once context is available
`
Probably right above the line that calls CoreRegistry.